pkg/apmintegration/setup.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (C) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package apmintegration
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/ava-labs/apm/apm"
+	"github.com/ava-labs/apm/config"
+	"github.com/ava-labs/avalanche-cli/pkg/application"
+	"github.com/ava-labs/avalanche-cli/pkg/constants"
+	"github.com/go-git/go-git/v5/plumbing/transport/http"
+	"github.com/spf13/afero"
+	"github.com/spf13/viper"
+	"gopkg.in/yaml.v2"
+)
+
+const (
+	credentialsFileKey  = "credentials-file"
+	adminAPIEndpointKey = "admin-api-endpoint"
+)
+
+// Note, you can only call this method once per run
+func SetupApm(app *application.Avalanche, apmBaseDir string) error {
+	credentials, err := initCredentials()
+	if err != nil {
+		return err
+	}
+
+	fs := afero.NewOsFs()
```

Comment:
please add comment on why we are doing this

Reply:
done